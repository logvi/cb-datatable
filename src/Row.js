// @flow
import React from 'react';
import type {RowProps} from './types';
import Cell from './Cell';

function Row({columns, row, onRowClick, className, ...rest}: RowProps) {
  return (
    <tr className={className} onClick={onRowClick ? e => onRowClick(e, row) : null} {...rest}>
      {columns.map((column, i) =>
        <Cell
          column={column}
          row={row}
          key={i} />
      )}
    </tr>
  );
}

export default Row;
