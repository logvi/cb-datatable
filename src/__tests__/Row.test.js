import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import Row from '../Row';

describe('Row', () => {
  it('Should have className', () => {
    const tree = shallow(
      <Row
        columns={[]}
        className="test"
      />
    );

    expect(tree.hasClass('test')).toBeTruthy();
  });

  it('Should pass a data prop', () => {
    const tree = renderer.create(
      <Row
        columns={[]}
        className="test"
        data-tooltip="tooltip"
        onMouseOver={() => {}}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  })
});
