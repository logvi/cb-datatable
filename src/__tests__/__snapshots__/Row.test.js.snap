// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Should pass a data prop 1`] = `
<tr
  className="test"
  data-tooltip="tooltip"
  onClick={null}
  onMouseOver={[Function]}
/>
`;
